crates/hir-ty/src/display.rs

```diff
@@ -323,10 +343,43 @@ impl HirDisplay for ProjectionTy {
             return write!(f, "{}", TYPE_HINT_TRUNCATION);
         }
 
+        let assoc_name =
+            f.db.type_alias_data(from_assoc_type_id(self.associated_ty_id)).name.to_string();
```

Comment:
`Name` implements `Display` and `PartialEq`, so can we avoid string allocation here (and L.367)?

Reply:
Yep. Somehow missed the fact that implements `PartialEq`, oops.